### DISCUSSION TITLE
`Question` : Change Default provider?

### DISCUSSION THREAD

**Discussion by OmiChrome** [gsuuon/llm.nvim#11] [Q&A] [answered]
i have simple Questions how do i change default provider to `PALM`, 
since its free and my openai api has expired, :(

**Comment by gsuuon** [comment, marked as answer]
Hi! You can set the `default_prompt` in [setup](https://github.com/gsuuon/llm.nvim#configuration). Just set it to a prompt that uses the PaLM provider - it's currently one of the [starters](https://github.com/gsuuon/llm.nvim/blob/f3a3881094389e6ad341fb955e55929c9d1f011a/lua/llm/prompts/starters.lua#L58), so you could do `default_prompt = require('llm.prompts.starters').palm`

**Reply by orhnk** [reply]
![grafik](https://github.com/gsuuon/llm.nvim/assets/101834410/5ba51e99-f0d9-4e6a-859d-0e0a904141b4)

```lua
{
    "gsuuon/llm.nvim",

    cmd = "Llm",

    keys = {
      { "<leader>al", "<cmd> Llm<CR>", mode = { "n", "v" }, desc = "LLM Generate" },
    },

    config = function(_, opts)
      require("llm").setup(opts)
    end,

    opts = function()
      return {
        default_prompt = require("llm.prompts.starters").palm,
        hl_group = "",
        prompts = {},
      }
    end,
  }
```

> [!NOTE]
> the PALM_API_KEY is set.
> #19

**Reply by OmiChrome** [reply]
this is what i use & it work's
```lua
  { -- multi language model free palm 2 api
    "gsuuon/llm.nvim",
    cmd = { "Llm", },
    config = function()
      require('llm').setup({
        default_prompt = require('llm.prompts.starters').palm,
        hl_group = 'Substitute',
      })
    end
  },
```

**Reply by orhnk** [reply]
Thanks @NormTurtle but mine is exactly same as yours.
The issue is different I think

**Reply by gsuuon** [reply]
@UTFeight  Are you sure you have `curl` installed and working? That error means curl didn't spawn correctly.
